homeassistant/components/holiday/config_flow.py

```diff
@@ -46,7 +46,12 @@ async def async_step_user(
 
             self._async_abort_entries_match({CONF_COUNTRY: user_input[CONF_COUNTRY]})
 
-            locale = Locale(self.hass.config.language)
+            try:
+                locale = Locale(self.hass.config.language)
+            except UnknownLocaleError:
+                # Default to (US) English if language not recognized by babel
+                # Mainly an issue with English flavors such as "en-GB"
+                locale = Locale("en")
```

Comment:
python-holidays uses BCP 47 language tags, so it is `en_GB` there instead of `en-GB`. Maybe you could just replace the hyphen with an underscore?

Reply:
Probably a good idea but I'll keep the exception around anyhow